contracts/governance/src/governance.rs

```diff
@@ -70,7 +75,10 @@ impl GovernanceContract {
         (!self.paused).then_some(()).ok_or(Error::ContractIsPaused)
     }
 
+    // to keep code consistent with other collections, we supress deref warnings
+    // as its not implemented for other when we switch features.
     fn is_allowed(&self, address: &PublicKey) -> Result<(), Error> {
+        #[allow(clippy::needless_option_as_deref)]
```

Comment:
As we discussed in the call, we don't need to switch between `map` and `vec` we can just stick on the one is more performant for the time being, and therefore remove this by adapting the code.